x-pack/plugins/monitoring/server/lib/alerts/fetch_index_shard_size.ts

```diff
@@ -134,15 +135,20 @@ export async function fetchIndexShardSize(
       }
       const {
         _index: monitoringIndexName,
-        _source: { index_stats: indexStats },
+        _source: { index_stats: indexStats, elasticsearch },
```

Comment:
if you just declare indexStats like this:
`const indexStats = topHit._source.index_stats || topHit._source.elasticsearch?.index`, I don't think you'd need to make any other changes below this line @klacabane

Reply:
This will make the 8.0.1 backport a bit of a pain in this file as well.